app/reports/subscription_collection_report.rb

```diff
@@ -0,0 +1,45 @@
+class SubscriptionCollectionReport
+  def initialize(subscriptions)
+    @subscriptions = subscriptions.map do |subscription|
+      SubscriptionReport.new(subscription, columns)
+    end
+  end
+
+  def to_csv
```

Comment:
So you need to rethink the way you're generating this report. You're using the `.generate` method that creates a string and this can cause memory issues. Also, you are returning the entire report through the app, which can cause issues to the web server.